interp/build.go

```diff
@@ -147,10 +147,22 @@ func skipFile(ctx *build.Context, p string, skipTest bool) bool {
 	}
 	a := strings.Split(p[i+1:], "_")
 	last := len(a) - 1
-	if last1 := last - 1; last1 >= 0 && a[last1] == ctx.GOOS && a[last] == ctx.GOARCH {
-		return false
+	if last-1 >= 0 {
+		switch x, y := a[last-1], a[last]; {
+		case x == ctx.GOOS:
+			if knownArch[y] {
+				return y != ctx.GOARCH
+			}
+			return false
+		case knownOs[x] && knownArch[y]:
+			return true
+		case knownArch[y] && y != ctx.GOARCH:
+			return true
+		default:
+			return false
+		}
```

Comment:
we could probably do better, but I didn't want to restart from scratch, so I modified just enough compared to your version to make the tests pass.

Reply:
Thanks for adding the missing tests and fixes. I can easily read and understand the logic of this code, so it is good for me!